codex-rs/tui/src/history_cell.rs

```diff
@@ -541,6 +541,39 @@ pub(crate) fn padded_emoji(emoji: &str) -> String {
     format!("{emoji}\u{200A}")
 }
 
+/// Format a clickable link with fallback for terminals that don't support hyperlinks
+fn format_link(url: &str, text: &str) -> String {
+    // Check if we're in a terminal that supports hyperlinks
+    if supports_hyperlinks() {
+        format!("\u{1b}]8;;{url}\u{7}{text}\u{1b}]8;;\u{7}")
+    } else {
+        // Fallback: show text with URL in parentheses
+        format!("{text} ({url})")
+    }
+}
+
+/// Detect if the terminal supports clickable hyperlinks
+fn supports_hyperlinks() -> bool {
+    // Check for known terminals that support hyperlinks
+    if let Ok(term) = std::env::var("TERM_PROGRAM") {
+        match term.as_str() {
+            "iTerm.app" | "vscode" | "Hyper" => return true,
+            "WarpTerminal" | "Warp" => return false, // Warp doesn't support clickable links yet
```

Comment:
Is there an issue tracking this missing feature in Warp? If there is line-of-sight to this getting fixed in Warp, I am less inclined to take this change.